src/contracts/GPv2AllowanceManager.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: LGPL-3.0-or-newer
+pragma solidity ^0.7.5;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+/// @title Gnosis Protocol v2 Allowance Manager Contract
+/// @author Gnosis Developers
+contract GPv2AllowanceManager {
+    using SafeERC20 for IERC20;
+
+    /// @dev A struct representing transfers to be executed as part of a batch
+    /// settlement.
+    struct Transfer {
+        address owner;
+        IERC20 token;
+        uint256 amount;
+    }
+
+    /// @dev The owner contract of the allowance manager. The owner is set at
+    /// creation time and cannot change.
+    address private immutable owner;
+
+    constructor() {
+        owner = msg.sender;
```

Comment:
Interesting decision, so you plan to create this contract from the constructor of `GPv2Settlement`? I'd have expected the owner to be part of the constructor input and having a separate deployment.

Reply:
Yes, that was the intention.

> I'd have expected the owner to be part of the constructor input and having a separate deployment.

If the settlement contract code gets too big, we might have to resort to that.